applicationinsights-react-native/src/ReactNativePlugin.ts

```diff
@@ -93,6 +109,7 @@ export class ReactNativePlugin extends BaseTelemetryPlugin {
             _self.setDeviceInfoModule = (deviceInfoModule: IDeviceInfoModule) => {
                 // Set the configured deviceInfoModule
                 _deviceInfoModule = deviceInfoModule;
+                _self._collectDeviceInfo();
```

Comment:
I don't think that this should be called just because it was set, unless there was some other bug that you found.

At the very least it should check the config.

Reply:
Will this corner case happened?
User call setDeviceInfoModule after init. As the config did not change and _deviceInfoModule is not referenced during the onConfig changes, the _self._collectDeviceInfo() would not be called and the info would not be updated.

Reply:
Yes, it's quite possible

Reply:
I added the _self._collectDeviceInfo(); here because I worried about this corner case. 
Another way would be referencing _deviceInfoModule in the onConfigChange. When user called setDeviceInfoModule, _self._collectDeviceInfo() would be automatically called and hence updated.